src/Symfony/Component/HttpFoundation/Request.php

```diff
@@ -1649,32 +1662,39 @@ public function getLanguages(): array
         $this->languages = [];
         foreach ($languages as $acceptHeaderItem) {
             $lang = $acceptHeaderItem->getValue();
-            if (str_contains($lang, '-')) {
-                $codes = explode('-', $lang);
-                if ('i' === $codes[0]) {
-                    // Language not listed in ISO 639 that are not variants
-                    // of any listed language, which can be registered with the
-                    // i-prefix, such as i-cherokee
-                    if (\count($codes) > 1) {
-                        $lang = $codes[1];
-                    }
-                } else {
-                    for ($i = 0, $max = \count($codes); $i < $max; ++$i) {
-                        if (0 === $i) {
-                            $lang = strtolower($codes[0]);
-                        } else {
-                            $lang .= '_'.strtoupper($codes[$i]);
-                        }
-                    }
-                }
-            }
+            $lang = $this->formatLocale($lang);
 
             $this->languages[] = $lang;
         }
 
         return $this->languages;
     }
 
+    private static function formatLocale(string $locale): string
+    {
+        if (str_contains($locale, '-')) {
```

Comment:
Now that you have moved this code to a new method, let's refactor it a bit by returning early if `!str_contains($locale, '-')`, thus removing one nested level.

Reply:
You are right, I will update this PR and remove a level of nesting.